metrics_proto_api_test.go

```diff
@@ -0,0 +1,1138 @@
+			inMetric: []*metricspb.Metric{
+				{
+					MetricDescriptor: inMDCall,
+					Timeseries:       []*metricspb.TimeSeries{},
+				},
+				{
+					MetricDescriptor: inMDLatency,
+					Timeseries:       []*metricspb.TimeSeries{},
+				},
+			},
+			outTSR: []*monitoringpb.CreateTimeSeriesRequest{
+				{
+					Name:       "projects/metrics_proto_test",
+					TimeSeries: []*monitoringpb.TimeSeries{},
+				},
+				{
+					Name:       "projects/metrics_proto_test",
+					TimeSeries: []*monitoringpb.TimeSeries{},
+				},
+				{
+					Name:       "projects/metrics_proto_test",
+					TimeSeries: []*monitoringpb.TimeSeries{},
+				},
+			},
+			outMDR: []*monitoringpb.CreateMetricDescriptorRequest{
+				{
+					Name:             "projects/metrics_proto_test",
+					MetricDescriptor: outMDCall,
+				},
+				{
+					Name:             "projects/metrics_proto_test",
+					MetricDescriptor: outMDLatency,
+				},
+			},
+		},
+	}
+	for i := 0; i < 500; i++ {
+		k := i / 250
+		v := fmt.Sprintf("value_%d", i)
+		lv := &metricspb.LabelValue{Value: v, HasValue: true}
+		ts := &metricspb.TimeSeries{
+			StartTimestamp: startTimestamp,
+			LabelValues:    []*metricspb.LabelValue{inEmptyValue, lv},
+			Points:         inPointsInt64_1,
+		}
+		tcs[0].inMetric[k].Timeseries = append(tcs[0].inMetric[k].Timeseries, ts)
+
+		j := i / 200
+		mt := outMetricTypeCalls
+		if k == 1 {
+			// TimeSeries Belongs to Latency
+			mt = outMetricTypeLatency
+		}
+		outTS := &monitoringpb.TimeSeries{
+			Metric: &googlemetricpb.Metric{
+				Type: mt,
+				Labels: map[string]string{
+					"empty_key":      "",
+					"operation_type": v,
+				},
+			},
+			Resource:   outGlobalResource,
+			MetricKind: googlemetricpb.MetricDescriptor_CUMULATIVE,
+			ValueType:  googlemetricpb.MetricDescriptor_INT64,
+			Points:     outPointsInt64_1,
+		}
+		tcs[0].outTSR[j].TimeSeries = append(tcs[0].outTSR[j].TimeSeries, outTS)
+	}
+	for _, tc := range tcs {
+		executeTestCase(t, tc, se, server)
+	}
+}
+
+func createConn(t *testing.T, addr string) *grpc.ClientConn {
+	conn, err := grpc.Dial(addr, grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Failed to make a gRPC connection to the agent: %v", err)
+	}
+	return conn
+}
+
+func createExporter(t *testing.T, conn *grpc.ClientConn, opts sd.Options) *sd.Exporter {
+	opts.MonitoringClientOptions = []option.ClientOption{option.WithGRPCConn(conn)}
+	se, err := sd.NewExporter(opts)
+	if err != nil {
+		t.Fatalf("Failed to create the statsExporter: %v", err)
+	}
+	return se
+}
+
+func executeTestCase(t *testing.T, tc *testCases, se *sd.Exporter, server *fakeMetricsServer) {
+	dropped, err := se.PushMetricsProto(context.Background(), nil, nil, tc.inMetric)
+	if dropped != 0 || err != nil {
+		t.Fatalf("Name: %s, Error pushing metrics, dropped:%d err:%v", tc.name, dropped, err)
+	}
+
+	var gotTimeSeries []*monitoringpb.CreateTimeSeriesRequest
+	server.forEachStackdriverTimeSeries(func(sdt *monitoringpb.CreateTimeSeriesRequest) {
```

Comment:
May be simpler to just do `gotTimeSeries := server.stackdriverTimeSeries`.

Reply:
done.